sympy/__init__.py

```diff
@@ -22,6 +22,21 @@
 
 from sympy.release import __version__
 
+if 'dev' in __version__:
+    def enable_warnings():
+        import os
+        import warnings
+        action ='default'
+        travis = (os.environ.get('TRAVIS','') == 'true')
```

Comment:
I think it would be better to always fail when the tests are running (or always). Things that are only reproducible in Travis are annoying.